spec/unit/inventory_spec.rb

```diff
@@ -51,4 +51,52 @@
       expect(resource.children_by_type('type-03', true).size).to eq(1)
     end
   end
+
+  describe '#ResultFetcher' do
+    it 'should fetch two first pages' do
+      batches = [[1, 2], [3, 4], [5]]
+      page_size = 2
+      calls_count = 0
+      fetch_func = lambda do |offset|
+        calls_count += 1
+        { 'startOffset' => offset, 'resultSize' => 5, 'results' => batches[offset / page_size] }
+      end
+      result_fetcher = Hawkular::Inventory::ResultFetcher.new(fetch_func)
```

Comment:
And this to a `subject` block.

Reply:
Can you detail?
From the docs here [1], subject is either used with `its` as a pojo (poro?) to evaluate its fields, or with method `Subject.should`. I can't see how it can be used here, when I need to call `Enumeration.take` with various arguments, or `Enumeration.collect`. How do you suggest I use it?

[1] https://relishapp.com/rspec/rspec-core/v/2-0/docs/subject